drools-compiler/src/main/java/org/drools/compiler/compiler/io/memory/MemoryFileSystem.java

```
@@ -170,11 +170,11 @@ public boolean existsFolder(String path) {
         if (path == null) {
             throw new NullPointerException("Path can not be null!");
         }
-        return folders.get( path.length() > 0 ? MemoryFolder.trimLeadingAndTrailing( path ) : path ) != null;
+        return folders.get(MemoryFolder.trimLeadingAndTrailing(path)) != null;
     }
 
     public boolean existsFile(String path) {
-        return fileContents.containsKey( MemoryFolder.trimLeadingAndTrailing( path ) );
+        return fileContents.containsKey(MemoryFolder.trimLeadingAndTrailing(path));
```

I'd say no (but this is opinion), not until we see a valid use case that pass a null value to this method. We can add a javadoc with "@param path never null" to clearly indicate the contract of this method.

A null check like that is "defensive code". Defensive code is great of public api's, but a performance consideration and maintenance worry for internal code. I strongly prefer just javadoccing params and return values as either "never null" or "sometimes null".

Furthermore, a null check can be handle in 2 ways: fail fast by throwing a clear IllegalState|ArgumentException (this should be your default behaviour!) or make the method null-safe. In this particular case, making the method null-safe is doable and reasonable (just return null), but in most cases it's not reasonable because it causes bug later down the line (without a correct error message or a outright misbehavior without an error message at all).